code/game/machinery/autodoc_excelsior.dm

```diff
@@ -0,0 +1,251 @@
+/obj/machinery/excelsior_autodoc
+	name = "Excelsior Autodoc"
+	desc = "Medical care for everybody, free, and may no one be left behind!"
+	icon = 'icons/obj/machines/excelsior/autodoc.dmi'
+	icon_state = "base"
+	density = TRUE
+	anchored = TRUE
+	circuit = /obj/item/electronics/circuitboard/excelsior_autodoc
+	use_power = IDLE_POWER_USE
+	idle_power_usage = 60
+	active_power_usage = 10000
+	var/mob/living/carbon/occupant
+	var/datum/autodoc/autodoc_processor
+	var/cover_closed = FALSE
+	var/cover_locked = FALSE
+	var/cover_moving = FALSE
+	var/image/screen_state = null
+	var/image/cover_state = null
+
+/obj/machinery/excelsior_autodoc/New()
+	. = ..()
+	autodoc_processor = new/datum/autodoc()
+	autodoc_processor.holder = src
+	autodoc_processor.damage_heal_amount = 20
+	update_icon()
+	
+
+/obj/machinery/excelsior_autodoc/Destroy()
+	if(occupant)
+		qdel(occupant)
+	return ..()
+
+/obj/machinery/excelsior_autodoc/relaymove(mob/user)
+	if (user.stat)
```

Comment:
you should probably also check for stunned here.

Reply:
Dunno, it worked fine for other machinery so far.